src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingUniversity.java

```diff
@@ -241,8 +244,7 @@ public void onSuccess(final ILocalResearch research)
         }
 
         LanguageHandler.sendPlayersMessage(colony.getMessagePlayerEntities(),
-          RESEARCH_CONCLUDED + random.nextInt(3),
-          IGlobalResearchTree.getInstance().getResearch(research.getBranch(), research.getId()).getDesc());
+          RESEARCH_CONCLUDED + random.nextInt(3), I18n.format(IGlobalResearchTree.getInstance().getResearch(research.getBranch(), research.getId()).getDesc()));
```

Comment:
We shouldn't use I18n.format, can't we just give it a TranslationTextComponent?

Reply:
I’ll take another shot at doing it that way, if it’s possible.  Was having a hell of a time with TranslationTextComponents not handling insert of a translated text into a %s, and instead doing an appended of translated text.

Reply:
Because formatting on the server side, afaik, will only ever return english translations to people.
You could try to wrap a translationcomponent inside a string component, that might work if the other doesn't.

Reply:
Looks like it's a part of the logic in https://github.com/ldtteam/Structurize/blob/version/1.16.3/src/main/java/com/ldtteam/structurize/util/LanguageHandler.java 's buildChatComponent to treat ITextComponents of any type as appends.  Probably easier to work around it for this case, ugly as the hack would be, than to revamp it there.

Reply:
Yeah, SendPlayersMessage is painful with it's behavior on TranslationTextComponent. 

If you look in AbstractBuildingWorker|ImproveRecipe you can see a similar chunk of code that is very simple and does the replacements the way you expect.